sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/entity_resolution/Commons.scala

```diff
@@ -16,17 +15,22 @@ import org.apache.spark.RangePartitioner
 import org.apache.spark.storage.StorageLevel
 import org.apache.spark.RangePartitioner
 import org.apache.spark.RangePartitioner
+import scala.io.Source
+import java.io.Serializable
 
 /* A generic Entity Resolution Approach for larger datasets(for e.g. 24.8GB)
  * Dataset1 has 47 million entities, approx 19GB
  * Dataset2 has 19 million entities, approx 5.8GB
  * */
-class EntityResolution_RDFData_CountVectorizer(spark: SparkSession, triplesRDD1: RDD[Triple], triplesRDD2: RDD[Triple],
-                                         teacher: DataFrame, threshold_subject: Double, jsimilarity_predicate: Double,
-                                         threshold_object: Double, vocab_size: Long, output_path: String) extends Serializable {
+abstract class Commons(val spark: SparkSession, val sourceData1: RDD[Triple], val sourceData2: RDD[Triple],
+                                         val teacher: DataFrame, val thresholdSubject: Double, val jsimilarityPredicate: Double,
```

Comment:
Why `teacher`? Are we making this generic enough which will accept any other type (RDF) dataset? I think we should remove such a misleading name. Of course, in your use-case, it makes totally sense, but here we should keep it more generic, as we aim to provide it as an API within SANSA.

Reply:
teacher is basically the labelled data to which we compare our predictions later on to evaluate recall and precision values. Yes, our algorithm can accept any other RDF dataset but n order to evaluate how well the entities are resolved, we would need the gold standards i.e. teacher in our case. I can change it to labels if you suggest.

Reply:
Right. I think you should include that DF as part of the algorithm. If you use validation data just for the sake of prediction and not training, I think we should generalize it to accept any other validation data. So, better remove it from the main algorithm and just add it when running the example.